packages/phoenix-event-display/src/loaders/objects/phoenix-objects.ts

```diff
@@ -277,11 +324,16 @@ export class PhoenixObjects {
       PhoenixObjects.sphericalToCartesian(4000, theta, clusterParams.phi)
     );
 
-    // if (cube.position.x * cube.position.x + cube.position.y * cube.position.y > maxR * maxR) {
-    //   cube.position.x = maxR * Math.cos(clusterParams.phi);
-    //   cube.position.y = maxR * Math.sin(clusterParams.phi);
-    // }
-    // cube.position.z = Math.max(Math.min(pos.z, maxZ), -maxZ); // keep in maxZ range.
+    // FIXME - more elegant way to do this? Maybe natively use cylindrical here?
+    // How to generalise? Pass in limit lambda?
+    if (
+      cube.position.x * cube.position.x + cube.position.y * cube.position.y >
+      maxR * maxR
```

Comment:
Not a more elegant way but using JavaScript's built-in methods, may be we can shorten it.

```ts
Math.pow(cube.position.x, 2) + Math.pow(cube.position.y, 2) > Math.pow(maxR, 2)
```

Reply:
By “elegant” I meant that this is currently hard coded for ATLAS and we should generalise. Not quite sure how to do it. I can change to use Math.pow but for simple squares I thought this was faster?